pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -9,9 +9,112 @@ import "google/protobuf/empty.proto";
 
 service ChainReader {
   rpc GetLatestValue (GetLatestValueRequest) returns (GetLatestValueReply) {}
+  rpc QueryKeys (QueryKeysRequest) returns (QueryKeysReply) {}
   rpc Bind(BindRequest) returns (google.protobuf.Empty) {}
 }
 
+message AndFilter {
+  repeated QueryKeysFilter filters = 1;
+}
+
+message AddressFilter {
+  repeated string addresses = 1;
+}
+
+message KeysFilter {
+  repeated string keys = 1;
+}
+
+message KeysByValueFilter {
+  repeated Keys keys = 1;
+  repeated Values values = 2;
+
+}
+
+message Keys {
+  repeated string keys = 1;
+}
+
+message Values {
+  repeated string values = 1;
+}
+
+message Confirmations {
+  int32 confirmation = 1;
+}
+
+message ConfirmationsFilter{
+  Confirmations confirmations = 1;
+}
+
+message ComparisonOperator {
+  int32 comparisonOperator = 1;
```

Comment:
what is this? should it be an enum of things like EQ, NOTEQ LT, GT, LTE, GTE?

Reply:
Yeah it should, just haven't done that yet

Reply:
added now